### PR TITLE
Correcting some grammatical errors of  README.md

### PR DESCRIPTION
**Pull request by vashisthabhinav**
Corrected some sentences by adding hyphens and commas wherever required and changed one sentence to make it grammatically more correct.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Read in different language : [**zh**](localization/zh/README.md), [**ko**](local
 
 # Introduction
 
-Design patterns are the best formalized practices a programmer can use to
+Design patterns are the best-formalized practices a programmer can use to
```
Comment (iluwatar) [inline README.md:24]: I think the intention here is to say `Design patterns are the best, formalized practices...`

File: README.md [hunk 2]
```diff
@@ -56,8 +56,8 @@ of the following approaches
  - Using tags such as `Performance`, `Gang of Four` or `Data access`.
  - Using pattern categories, `Creational`, `Behavioral`, and others.
 
-Hopefully you find the object oriented solutions presented on this site useful
-in your architectures and have as much fun learning them as we had developing them.
+Hopefully, you find the object-oriented solutions presented on this site useful
+in your architectures and have as much fun learning them as we had when we were developing them.
```
Comment (iluwatar) [inline README.md:60]:
```suggestion
in your architectures and have as much fun learning them as we had while developing them.
```

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vashisthabhinav) [reply README.md:24]: Ok correcting.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
@all-contributors please add @vashisthabhinav for doc

**Comment by iluwatar** [comment]
Thank you for the enhancements @vashisthabhinav

**Pull request merged**